docs/pages/textarea.js

```diff
@@ -36,219 +53,270 @@ export default function DocsPage({ generatedDocGen }: {| generatedDocGen: DocGen
-      label="With a placeholder"
-      value={value}
-    />
-  );
-}
-`}
-      />
-      <Example
-        id="helperText"
-        name="Example: Helper Text"
-        description={`Whenever you want to provide more information about a form field, you should use \`helperText\`.`}
-        defaultCode={`
-function Example(props) {
-  const [value, setValue] = React.useState('')
-  return (
-    <Box padding={2} color="white">
-      <TextArea
-        id="aboutmemore"
-        onChange={({value}) => setValue(value)}
-        placeholder="Write something about yourself..."
-        helperText="I love to sail, run and visit remote places"
-        label="With a placeholder"
-        value={value}
-      />
-    </Box>
-  );
-}
-`}
-      />
-      <Example
-        id="errorMessageExample"
-        name="Example: Error message"
-        description={`
+      <MainSection name="Variants">
+        <MainSection.Subsection
+          title="Basic"
+          description={`
+      A \`TextArea\` will expand to fill the width of the parent container.
+    `}
+        >
+          <MainSection.Card
+            defaultCode={`
+  function Example(props) {
+    const [value, setValue] = React.useState('')
+    return (
+      <Box width="100%">
+        <TextArea
+          id="aboutme"
+          onChange={({value}) => setValue(value)}
+          placeholder="Write something about yourself..."
+          label="About me"
+          value={value}
+        />
+      </Box>
+    );
+  }
+      `}
+          />
+        </MainSection.Subsection>
+
+        <MainSection.Subsection
+          title="Helper text"
+          description={`Whenever you want to provide more information about a form field, you should use \`helperText\`.`}
+        >
+          <MainSection.Card
+            defaultCode={`
+  function Example(props) {
+    const [value, setValue] = React.useState('')
+    return (
+      <Box width="100%">
+        <TextArea
+          id="aboutmemore"
+          onChange={({value}) => setValue(value)}
+          placeholder="Write something about yourself..."
+          helperText="Describe your favorite hobbies, foods, or books."
+          label="About me"
+          value={value}
+        />
+      </Box>
+    );
+  }
+      `}
+          />
+        </MainSection.Subsection>
+        <MainSection.Subsection
+          title="Readonly"
+          description={`
+      A \`TextArea\` can be in read-only mode in order to present information to the user without allowing them to edit the content. Typically they are used to show content or information that the user does not have permission or access to edit.
+    `}
+        >
+          <MainSection.Card
+            defaultCode={`
+  function Example(props) {
+    const [value, setValue] = React.useState('To keep shopping inspirational and actionable, we set high standards for our Merchants. Your website was not approved due to fuzzy, low quality images.');
```

Comment:
nit: this doesn't really seem like a realistic use case for read-only TextArea, but I also don't have a better one so feel free to run with this for now if you like. but it doesn't strike me as something we'd display in a TextArea

Reply:
Yeah, it's the example from the person in Slack, but they have it in an error modal which makes it feel more "correct". I'll go with this for now and see if we can find a better product example.